packages/babel-plugin-proposal-object-rest-spread/test/fixtures/object-spread/no-object-assign-exec/exec.js

```diff
@@ -0,0 +1,32 @@
+Object.defineProperty(Object.prototype, 'NOSET', {
+  set(value) {
+    // noop
+  },
+});
+
+Object.defineProperty(Object.prototype, 'NOWRITE', {
+  writable: false,
+  value: 'abc',
+});
+
+const obj = { NOSET: 123 };
+// this wouldn't work as expected if transformed as Object.assign (or equivalent)
+// bacause those trigger object setters (spread don't)
+const objSpread = { ...obj };
+
+const obj2 = { NOSET: 123, NOWRITE: 456 };
+// this line would throw `TypeError: Cannot assign to read only property 'NOWRITE'`
+// if transformed as Object.assign (or equivalent) because those use *assignment* for creating properties
+// (spread defines them)
+const obj2Spread = { ...obj2 };
+
+assert.deepEqual(obj, objSpread);
+assert.deepEqual(obj2, obj2Spread);
+
+const obj3Spread = { ...{ get foo () { return 'bar' } } };
```

Comment:
I've also added those tests that @Kovensky suggested they would fail (now passing) 👌